### PR TITLE
Two new features

### PR DESCRIPTION
**Pull request by ekengni**
Feature 1: Requesting Number of comments 

- Client can request the number of comments to be requested from the datastore.

Feature 2: Deleting comments

- Client can choose to delete all the comments in the database.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -53,29 +77,30 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
   /** Converts message of arrays into json fromat */
-  /** Deprecated */
   private String convertToJsonUsingGson(){
     String json = new Gson().toJson(comments);
     return json;
   }
+
+  /**Returns the number of comments the user wants to display, or -1 if number was invalid */
+  private int getCommentLimit(HttpServletRequest request, int num_entities){
+    String comment_limit_string = request.getParameter("num-comments");
```
Comment (kavinkarthik92) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: Move magic string to a const?

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -29,13 +30,36 @@
...
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   static final String comment_id = "comment-input";
   static final String redirect_location = "/comment.html";
   private List <String> comments;
+  
+  /** Responsible for creating new comments */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
```
Comment (lazau-g) [inline src/main/java/com/google/sps/servlets/DataServlet.java]: Here and throughout the file, use string constants, e.g. `public static final String DATASTORE_MESSAGE = "Message";` for these types of strings.

For small projects, it's ok to repeat these string comments `"timestamp"`, `"Message"`, `"comment"`, `"num-comments"`, etc. However, as projects gains more collaborators, the chances of a typo increases. These typos creates hard to track down bugs. Additionally, a descriptive string constant name improves the readability of the code.

File: src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting comments. */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet{
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (lazau-g) [inline src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: Nitpick: DELETE, implemented with `doDelete`, is a better HTTP verb for deleting data.

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -39,27 +39,63 @@ function createMap() {
...
   const liElement = document.createElement('li');
-  liElement.innerText = text;
+  liElement.innerText = message;
+
+  /*headElement = document.createElement('h2');
+  const headElement.innerTex; */
   return liElement;
 }
 
+function deleteListElement(messages, messagesListElement) {
```
Comment (lazau-g) [inline src/main/webapp/script.js:76]: It makes more sense to just make the delete button a plain HTML element which sends an HTTP DELETE (or HTTP POST) to the server. In response, the server redirects the browser to the same page with the updated comments list (containing no comments).

This approach reduces the amount of JS, and makes the button more straightforward as it's just in plain HTML.

**Review by lazau-g — CHANGES_REQUESTED** [review]
The delete button needs some work. You have the right idea, but there's a slightly more straightforward approach!

**Review by kavinkarthik92 — COMMENTED** [review]
I see that a lot of FrontEnd logic (HTML tags, CSS) is spilled over to the Backend (i,e the JAVA) code. Good practice is to keep them separate. In most projects, FrontEnd engineers would be working independent of Backend Engineers and hence any changes which the FrontEnd engineer makes should not break the work of Backend engineer and vice versa.

Give some thought and try to come up with ways on how you can separate both. Reach out to me if you have more doubts.

File: src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 3 above)
Comment (ekengni) [reply src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: oh I didn't know DELETE was a thing. Thank you.

File: src/main/webapp/script.js (hunk 4 above)
Comment (ekengni) [reply src/main/webapp/script.js:76]: Sure, but I wanted to have the delete button only appear when the comments appeared. It was more of a stylistic choice.

**Review by lazau-g — APPROVED** [review]

**Pull request merged**